docs/ingestion/automatic-compaction.md

```diff
@@ -0,0 +1,196 @@
+* Set `skipOffsetFromLatest` to reduce the chance of conflicts between ingestion and compaction.
+* Increase the priority value of compaction tasks relative to ingestion tasks. In the auto-compaction configuration, set `taskPriority` to the desired priority value. See [Lock priority](../ingestion/tasks.md#lock-priority) for the priority values of different task types.
+
+
+### Set frequency of compaction runs
+
+If you want the Coordinator to check for compaction more frequently than its indexing period, create a separate duty group and set its time period in the `coordinator/runtime.properties` file. For example, to change the auto-compaction period to 1 minute:
+```
+druid.coordinator.dutyGroups=["compaction"]
+druid.coordinator.compaction.duties=["compactSegments"]
+druid.coordinator.compaction.period=PT60S
+```
+
+## View automatic compaction statistics
+
+After the Coordinator has initiated auto-compaction, you can view compaction statistics for the datasource, including the number of bytes, segments, and intervals already compacted and those awaiting compaction. The Coordinator also reports the total bytes, segments, and intervals not eligible for compaction in accordance with its [segment search policy](../design/coordinator.md#segment-search-policy-in-automatic-compaction).
+
+In the Druid console, the Datasources view displays auto-compaction statistics. The Tasks view shows the task information for compaction tasks that were triggered by the automatic compaction system.
+
+To get statistics by API, send a [`GET` request](../operations/api-reference.md#get-10) to `/druid/coordinator/v1/compaction/status`. To filter the results to a particular datasource, pass the datasource name as a query parameter to the request—for example, `/druid/coordinator/v1/compaction/status?dataSource=wikipedia`.
+
+## Examples
+
+The following examples demonstrate potential use cases in which auto-compaction may improve your Druid performance. Note that compaction in itself does not modify the underlying data of the segments. However, you can use manual compaction tasks or the automatic compaction system to modify data to improve query performance. See more details in [Compaction strategies](../ingestion/compaction.md#compaction-strategies). The examples in this section do not change the underlying data.
+
+### Change segment granularity
+
+You have a stream set up to ingest data with `HOUR` segment granularity into the `wikistream` datasource. You notice that your Druid segments are smaller than the [recommended segment size](../operations/segment-optimization.md) of 5 million rows per segment. You wish to automatically compact segments to `DAY` granularity while leaving the latest week of data _not_ compacted as your stream may ingest data over that time period.
```

Comment:
```suggestion
You have a stream set up to ingest data with `HOUR` segment granularity into the `wikistream` datasource. You notice that your Druid segments are smaller than the [recommended segment size](../operations/segment-optimization.md) of 5 million rows per segment. You wish to automatically compact segments to `DAY` granularity while leaving the latest week of data _not_ compacted as your stream consistently has data over that time period coming in.
```

nit: I wonder if we want to mention that this is a problem only if the stream has consistently late arriving data for that time period. If there is late arriving data some of the time, that's not too much of a problem. The system is robust enough to handle that.

Reply:
will add a note below the example